app/lib/meetup.rb

```diff
@@ -0,0 +1,95 @@
+class Meetup
+  include HTTParty
+  base_uri 'api.meetup.com'
+
+  attr_reader :options
+
+  def initialize
+    api_key  = OperationCode.fetch_secret_with(name: :meetup_api_key)
+    @options = {
+      query: {
+      key: api_key,
+      sign: "true"
+      }
+    }
+  end
+
+  # GET all Operation Code Pro data. Returns all op-code groups.
+  def operationcode_data
+    response = self.class.get("/pro/operationcode/groups", options)
+    return_value_for response
+  end
+
+  # GET events for each group
+  def get_events_for(urlname)
+    response = self.class.get("/#{urlname}/events", options)
+    return_value_for response
+  end
+
+  #Return all meetups based on group names if venue is present. Return array of all op-code events
+  def event_details_by_group
+    events = []
+    group_names.each do |group|
+      get_events_for(group).each do |event|
+        next unless event['venue'].present?
+        events << details_for(event)
+      end
+    end
+    events.flatten
```

Comment:
Trying to stub event_details_by_group to better test add_events_to_database!  Want to test update cases and duplicates.  event_details_by_group returns an array of hashes.